src/import-export/csv-imp/gnc-tx-import.cpp

```diff
@@ -0,0 +1,960 @@
+{
+    auto parsed_date = parse_date (str.c_str(), date_format);
+    if (parsed_date == -1)
+        return nullptr;
+    else
+    {
+        auto mydate = new time64;
+        *mydate = parsed_date;
+        return mydate;
+    }
+}
+
+
+/** Convert str into a gnc_numeric using the user-specified (import) currency format.
+ * @param str The string to be parsed
+ * @param currency_format The currency format to use.
+ * @return a pointer to a gnc_numeric on success, nullptr on failure
+ */
+static gnc_numeric* convert_amount_col_str (const std::string &str, int currency_format)
+{
+    /* If a cell is empty or just spaces return 0 as amount */
+    if(!boost::regex_search(str, boost::regex("[0-9]")))
+        return nullptr;
+
+    auto expr = boost::make_u32regex("[[:Sc:]]");
+    std::string str_no_symbols = boost::u32regex_replace(str, expr, "");
+
+    /* Convert based on user chosen currency format */
+    gnc_numeric val;
+    char *endptr;
+    switch (currency_format)
+    {
+    case 0:
+        /* Currency locale */
+        if (!(xaccParseAmount (str_no_symbols.c_str(), TRUE, &val, &endptr)))
+            return nullptr;
+        break;
+    case 1:
+        /* Currency decimal period */
+        if (!(xaccParseAmountExtended (str_no_symbols.c_str(), TRUE, '-', '.', ',', "\003\003", "$+", &val, &endptr)))
+            return nullptr;
+        break;
+    case 2:
+        /* Currency decimal comma */
+        if (!(xaccParseAmountExtended (str_no_symbols.c_str(), TRUE, '-', ',', '.', "\003\003", "$+", &val, &endptr)))
+            return nullptr;
+        break;
+    }
+
+    auto amount = new gnc_numeric;
+    *amount = val;
+    return amount;
+}
+
+/* Define a class hierarchy to temporarily store transaction/split properties
+ * found in one import line. There is a generic parent class and an implementation
+ * template class. This template class is further specialized for each data type
+ * we support (currently time64, Account, string and gnc_numeric).
+ */
+struct GncTransProperty
+{
+    virtual ~GncTransProperty()
+      //Remove pure designation.
+        {}
+    bool m_valid = false;
+};
+
+template<class T>
+struct GncTransPropImpl
+: public GncTransProperty
+{
+public:
+    ~GncTransPropImpl(){};
+    GncTransPropImpl(const std::string& val, int fmt)
+    {
+        m_valid = false;
+    };
+
+    static GncTransProperty* make_new(const std::string& val,int fmt = 0)
+        { return nullptr; }
+
+    T value;
+};
+
+template<>
+struct GncTransPropImpl<time64*>
+: public GncTransProperty
+{
+    GncTransPropImpl(const std::string& val, int fmt)
+    {
+        value = convert_date_col_str (val, fmt);
+        m_valid = (value != nullptr);
+    }
+    ~GncTransPropImpl()
+        { if (value) delete value; }
+
+    static std::shared_ptr<GncTransProperty> make_new(const std::string& val,int fmt)
+    { return std::shared_ptr<GncTransProperty>(new GncTransPropImpl<time64*>(val, fmt)); }
+
+    time64* value;
```

Comment:
I don't think the compiler was complaining. I think it's a left-over from one of the many different ways I tried to get this to work. I believe it can just go away now.